quipucords/quipucords
# UI Issues/1710 update for pausing and cancelling scans

### What's included
- fix(ui scans): 1710 pause scans 
  * converted to using job ids instead of original scan ids
  * don't allow scan modifiers when pending
  * restructure scans to remove legacy React, componentWillReceiveProps
  * scansReducer, date now removed in favor of timestamp from response
  * scansReducer, test & snapshots updated
  * build, local run staging npm script
  * build, api.sh docker updates, db container

#### Additional convenience updates
The ui local run scripts  were updated based on the updates to the container structure
  * local run staging npm script
  * api.sh docker updates, db container

### Notes
Couple of quirks
- There's still some quirkiness around initial scans and connections failing, not sure that directly relates to this issue could just be network. It'll return one failed, and one successful job.
- In order to avoid issues and stay in alignment with the API when "pending" job state pops up we removed the ability to "pause"
  - CONFIRMED, there is a blend of status that's happening when a scan starts up between the detail message saying "[i'm running]" but the status is "[i'm pending]" ... based on this the exposed pause functionality is being hidden, so the GUI is technically working as intended. If we need to update the messaging to avoid confusion, can do 
![screen shot 2019-01-25 at 11 10 12 am](https://user-images.githubusercontent.com/3761375/51759096-6fd5a880-2095-11e9-82c3-50241936bae5.png)
 ~~there appears as if some blend between a job "running" and "pending" happens right after a job is resumed from a "pause". Either the GUI isn't updating or there's something at the API level. In this scenario the GUI only displays the "cancel" job option as if it's "pending". Continuous refreshes of the page don't seem to update the GUI.~~ Another set of eyes would help on this quirk.
- ~~appears to be a remaining issue of... need confirmation~~ @himmAllRight 
  * ~~pause a scan~~
  * ~~restart it~~
  * ~~try to pause it again, and it fails. unclear if this requires a new job id~~

### How to test
  1. Add a source, make sure you created a credential
  1. Create and start a scan
  1. Pause the scan
  1. Restart the scan
  1. Cancel the scan

### Examples
Cancel & Resume
![screen shot 2019-01-24 at 11 47 46 am](https://user-images.githubusercontent.com/3761375/51695190-6d5d4b00-1fd0-11e9-82fe-391bac3a8a7c.png)


Pause
![screen shot 2019-01-24 at 11 48 36 am](https://user-images.githubusercontent.com/3761375/51695072-2f602700-1fd0-11e9-924c-8b307be04756.png)


### Updates issue/story
Update #1710
